### ISSUE TITLE
`BNGUtils.pm` has `# TODO use strict;` — foundational module runs without strict checking

### ISSUE THREAD

**Issue by akutuva21** [RuleWorld/bionetgen#299]
## Summary

`BNGUtils.pm` — arguably the most foundational utility module in the BioNetGen Perl codebase — has been running without `use strict` since its creation. Line 2 of the file reads:

```perl
# TODO use strict;
```

This means typos in variable names silently create new variables, undeclared variables don't raise errors, and symbolic references are unchecked throughout all 27 subroutines in this module.

## Location

**File:** `bng2/Perl2/BNGUtils.pm`, line 2

Additionally, `ComponentType.pm` and `HNauty.pm` lack `use strict` entirely (not even a TODO comment).

## Current behavior

```perl
package BNGUtils;
# TODO use strict;

# Perl Modules
use FindBin;
```

## Impact

Without `use strict`:
- A typo like `$reuslt` instead of `$result` silently creates a new variable initialized to `undef`
- Variables can be used without declaration via `my`, making scope bugs invisible
- Symbolic references (e.g., using a string as a variable name) are allowed, which can cause hard-to-trace errors

Since `BNGUtils` provides core utilities used across the entire codebase (error handling, version info, math helpers, file I/O wrappers), any latent bugs here propagate everywhere.

## How to verify

```bash
head -5 bng2/Perl2/BNGUtils.pm
# Also check which other files lack use strict:
grep -rL 'use strict;' bng2/Perl2/*.pm
```

## Suggested fix

Uncomment and enable strict:

```perl
package BNGUtils;
use strict;
use warnings;
```

This will likely surface undeclared variable warnings that need to be fixed, but that's the point — those are latent bugs.

**Comment by akutuva21** [comment]
Closing the original BNGUtils.pm report because that module now enables both use strict and use warnings, addressing the undeclared-variable and scope-checking gap described here. ComponentType.pm, HNauty.pm, and Map.pm are separate cleanup scope and can be tracked independently; they do not leave the reported BNGUtils defect open.
